src/playsim/p_enemy.cpp

```diff
@@ -2592,6 +2608,7 @@ void A_DoChase (AActor *actor, bool fastchase, FState *meleestate, FState *missi
 			}
 			actor->flags7 &= ~MF7_INCHASE;
 			actor->goal = newgoal;
+			GoalReached(actor, savedgoal);
```

Comment:
Missing destruction check before calling this since a goal special could delete it.
`if (!(actor->ObjectFlags & OF_EuthanizeMe))`